qcore/siteamp_models.py

```diff
@@ -321,24 +325,42 @@ def ba18_amp(
     fhightop=100.0,
     fmax=15.0,
 ):
+    """
+
+    :param dt:
+    :param n:
+    :param vref: Reference vs used for waveform
+    :param vs: Actual vs30 value of the site
+    :param vpga: Reference vs for HF
+    :param pga: PGA value from HF
+    :param version: unused
+    :param flowcap: unused
+    :param fmin:
+    :param fmidbot:
+    :param fmid:
+    :param fhigh:
+    :param fhightop:
+    :param fmax:
+    :return:
+    """
     if vs > 1000:
         vs = (
             999
         )  # maximum vs30 supported by the model is 999, so caps the vsite to that value
 
-    ref, __ = ba_18_site_response_factor(vref, pga)
-    vsite, freqs = ba_18_site_response_factor(vs, pga)
+    ref, __ = ba_18_site_response_factor(vref, pga, vpga)
+    vsite, freqs = ba_18_site_response_factor(vs, pga, vpga)
 
     amp = np.exp(vsite - ref)
-    # interpolation function expects frequencies in descending order
-    ampv, ftfreq = interpolate_frequency(freqs[::-1], amp[::-1], dt, n)
+    ftfreq = get_ft_freq(dt, n)
 
-    ampf = amp_bandpass(ampv, fhightop, fmax, fmidbot, fmin, ftfreq)
+    ampi = np.interp(ftfreq, freqs, amp)
+    # ampfi = amp_bandpass(ampi, fhightop, fmax, fmidbot, fmin, ftfreq)
```

Comment:
I think removing the high frequency taper this way doesn't quite work because:
1. Now it doesn't apply the low frequency taper (using fmidbot and fmin).
2. The function amp_bandpass actually adds in another entry into the array at the start (i.e. line 174 makes a vector of ones that is one entry longer, and then at line 176 the vector gets populated starting from index 1). This makes ampi one entry shorter than ampfi.

My suggestion is to uncomment this and change the returned variable back to ampfi. For this line, we just make fhightop =999 and fmax=1000 or something like that.

Reply:
Just in case there's an misinterpretation, we should do fhightop=999 and fmax=1000 for line 358 only (i.e. only for BA18 site amp). Keep the amp_bandpass general because CB14 will still need it the way it is.